src/commands/api/getDatabase.ts

```diff
@@ -3,10 +3,33 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { DocDBDatabaseTreeItem } from '../../docdb/tree/DocDBDatabaseTreeItem';
+import { AzureParentTreeItem } from 'vscode-azureextensionui';
 import { ext } from '../../extensionVariables';
 import { MongoDatabaseTreeItem } from '../../mongo/tree/MongoDatabaseTreeItem';
+import { CosmosDBDatabase } from '../../vscode-cosmosdb.api';
 
-export async function getDatabase(): Promise<String> {
-    return (await ext.tree.showTreeItemPicker([MongoDatabaseTreeItem.contextValue, DocDBDatabaseTreeItem.contextValue])).fullId;
+export async function getDatabase(detectionData: { connectionString: string }): Promise<CosmosDBDatabase | undefined> {
+    const attachedAccountsNode = <AzureParentTreeItem | undefined>(await ext.tree.getChildren()).find((subscription) => {
+        return (subscription.id === 'cosmosDBAttachedAccounts');
+    });
+    if (attachedAccountsNode) {
+        const attachedAccounts = await attachedAccountsNode.getCachedChildren();
+        for (const account of attachedAccounts) {
+            if (account instanceof AzureParentTreeItem) {
+                const databases = await account.getCachedChildren();
+                for (const database of databases) {
+                    if (database instanceof MongoDatabaseTreeItem && database.connectionString === detectionData.connectionString) {
+                        return {
+                            accountName: account.label,
```

Comment:
You don't actually need to get a node in the tree. You just need to leverage some existing functions (probably `getServerIdFromConnectionString` and `getDatabaseNameFromConnectionString`) to get the accountName and databaseName